lib/http2/hpack.c

```diff
@@ -595,6 +601,79 @@ int h2o_hpack_parse_request(h2o_mem_pool_t *pool, h2o_hpack_decode_header_cb dec
     Next:;
     }
 
+    if (protocol.len > 0 &&
+                h2o_memis(method->base, method->len, H2O_STRLIT("CONNECT")) &&
+                (h2o_memis(protocol.base, protocol.len, H2O_STRLIT("connect-udp")) || h2o_memis(protocol.base, protocol.len, H2O_STRLIT("CONNECT-UDP"))) &&
+                (*scheme == &H2O_URL_SCHEME_HTTPS)) {
+        const char well_known_masque_udp_str[] = "/.well-known/masque/udp/"; // see: https://www.rfc-editor.org/rfc/rfc9298.html#section-3.4
+        size_t well_known_masque_udp_str_len = sizeof(well_known_masque_udp_str)-1;
+
+
+/*
```

Comment:
This comment block can be removed - it's here for explanation of the changes only